### PR TITLE
✨ Add ability to use as stand-alone pre-commit hook

### PR DESCRIPTION
**Pull request by s-weigand**
Hey thanks for the cool project ❤️ 

Since I want to replace `darglint` in my `pre-commit` configs with your project (so tired of waiting on docs to lint) it would be nice to integrate it as a stand-alone hook rather than adding it to `flake8` (my current workaround). 

Especially since I want to replace `flake8` with `ruff` anyway and `darglint` was the last big blocker.

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-   id: pydoclint
+    name: pydoclint
+    description: 'A Python docstring linter matching signatures.'
```
Comment (jsh9) [inline .pre-commit-hooks.yaml:3]:
```suggestion
    description: 'A Python docstring linter that checks arguments, returns, yields, and raises sections'
```

File: README.md [hunk 2]
```diff
@@ -58,7 +58,22 @@ flake8 --select=DOC <FILE_OR_FOLDER>
...
-### 2.3. Native vs _flake8_
+### 2.3. As a pre-commit hook
+
+`pydoclint` is configured for [pre-commit](https://pre-commit.com/)
+and can be set up as a hook with the following `.pre-commit-config.yaml`
+configuration:
+
+```yaml
+- repo: https://github.com/jsh9/pydoclint
+  rev: 0.0.7
```
Comment (jsh9) [inline README.md:69]: If we put `0.0.7` here, do we need to update this number here every time we publish a new release?  I wonder what other pre-commit hooks do in their README.

File: README.md [hunk 3]
```diff
@@ -58,7 +58,22 @@ flake8 --select=DOC <FILE_OR_FOLDER>
...
+  hooks:
+    - id: pydoclint
+```
+
+You will need to install `pre-commit` and run `pre-commit install`.
+
+### 2.5. Native vs _flake8_
```
Comment (jsh9) [inline README.md:76]:
```suggestion
### 2.4. Native vs _flake8_
```

File: README.md [hunk 4]
```diff
@@ -58,7 +58,22 @@ flake8 --select=DOC <FILE_OR_FOLDER>
...
+
+`pydoclint` is configured for [pre-commit](https://pre-commit.com/)
+and can be set up as a hook with the following `.pre-commit-config.yaml`
+configuration:
+
+```yaml
+- repo: https://github.com/jsh9/pydoclint
+  rev: 0.0.7
+  hooks:
+    - id: pydoclint
```
Comment (jsh9) [inline README.md:71]: Could you show an example or two on how to specify options in the pre-commit hook?

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -20,3 +20,8 @@ repos:
     hooks:
       - id: end-of-file-fixer
       - id: trailing-whitespace
+
+  - repo: https://github.com/pre-commit/pre-commit
+    rev: v3.2.2
+    hooks:
+      - id: validate_manifest
```
Comment (jsh9) [inline .pre-commit-config.yaml:27]: Curious question: what does this hook do?  I know it's an "official" pre-commit hook, but when I tried to search for its documentation, I couldn't find any.

Thanks!

**Comment by jsh9** [comment]
Hi @s-weigand , thank you so much for creating this PR!

I added some comments, and merged in the latest changes in `main` for you.

You mentioned that you'd like to use pydoclint as a standalone hook rather than with `flake8`.  That's fine, but please note that if you used it outside `flake8`, you wouldn't be able to use in-line omission (`# noqa: DOCxxx`): https://github.com/jsh9/pydoclint/blob/f2ffab506ad28bff9436e0d45f09b2228801026b/README.md

File: .pre-commit-config.yaml (hunk 5 above)
Comment (s-weigand) [reply .pre-commit-config.yaml:27]: That hook [lints the hook manifest itself](https://github.com/pre-commit/pre-commit/blob/c716de12f7533dc8ce522afb0c09f22476243f08/.pre-commit-hooks.yaml#L3).
So if a repo defines a hook that hook should also be checked to be valid, especially considering future compatibility.

File: README.md (hunk 2 above)
Comment (s-weigand) [reply README.md:68]: There are multiple ways to go about this
- Use a branch name like `main` as rev
- Use a generic non working term as template like `<latest_tag>`
- Use renovate to generate update PRs with new version ([example config for a nodejs hook](https://github.com/renovatebot/pre-commit-hooks/blob/main/renovate.json))

File: README.md (hunk 4 above)
Comment (s-weigand) [reply README.md:70]: Added an example of passing the config via args.

File: .pre-commit-config.yaml (hunk 5 above)
Comment (jsh9) [reply .pre-commit-config.yaml:27]: Ah I see.  Thank for the explanation!

File: README.md (hunk 2 above)
Comment (jsh9) [reply README.md:68]: Thanks!  I think the 2nd option is cleaner. I'll push a code change.

**Comment by jsh9** [comment]
Hi @s-weigand , thanks again!  I'm merging this PR as "fixed".

**Pull request merged**